src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -102,6 +102,29 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
 
             var originalPackageSpec = matchingPackageSpecs.FirstOrDefault();
 
+            // Convert relative path to absolute path if there is any
+            List<string> sourcePaths;
+
+            if (packageReferenceArgs.Sources?.Any() == true)
+            {
+                sourcePaths = new List<string>();
+                string projectDirectory = Path.GetDirectoryName(packageReferenceArgs.ProjectPath);
+
+                foreach (string source in packageReferenceArgs.Sources)
+                {
+                    sourcePaths.Add(UriUtility.GetAbsolutePath(Environment.CurrentDirectory, source));
```

Comment:
`dotnet add package` works in a peculiar way. We can verify that it works correctly by adding an extra integration test. 

Some background first. 
Read https://microsoft.sharepoint.com/teams/NuGet/_layouts/OneNote.aspx?id=%2Fteams%2FNuGet%2FTeam%2FNuGet&wd=target%28Engineering.one%7C219AABAA-18B3-46F4-83E8-D3188D83DE22%2FWhere%28%21%5C%29%20is%20NuGet%3F%7CC79766FE-842B-467F-BA94-438E51F8523C%2F%29
onenote:https://microsoft.sharepoint.com/teams/NuGet/Team/NuGet/Engineering.one#Where(!)%20is%20NuGet&section-id={219AABAA-18B3-46F4-83E8-D3188D83DE22}&page-id={C79766FE-842B-467F-BA94-438E51F8523C}&end. 

* NuGet ships NuGet.Commandline.Xplat in the .NET SDK, and commands such as `dotnet add package` are implemented by forwarding certain arguments. 
* Basically, there's a parsing on `dotnet.exe` process side, and then that's forwarded to `NUGet.Commandline.Xplat` that right now, repeats it's own parsing. 
* Environment.CurrentDirectory is the working directory that's passed at process start-up. Meaning the directory being read here is the one for `NuGet.Commandline.Xplat`. I am *not* sure if that is forwarded by the `dotnet.exe` implementation, but I'd expect that it is. 

You should make sure that it's resolve *against* the original startup directory and not relative to something like `nuget.commandline.xplat`'s start-up directory. 

That being said, `Dotnet.Integration.Tests` are the ones that run `dotnet.exe`, so I think adding a test in there tests the relative path is a good idea.

Reply:
Ok. I added source relative path unit tests into `Dotnet.Integration.Test`.
But I added source absolute path unit tests into Xplat.FuncTest, because previous one is slow than this one and I don't want to make it more slow by adding tests that can run faster elsewhere.